### PR TITLE
Generate map per round

### PR DESCRIPTION
**Pull request by Schtekt**
The map has earlier only been generated once per game, at the start.
This pull request generates a new map any time the gamescene is initiated.

Some changes had to be made to dynamic entities because of this. Th bbc adds itself to physics on init for instance.

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 1]
```diff
@@ -233,7 +233,7 @@ Entity* EnemyFactory::Add(const std::string& entityName, EnemyComps* comps)
 		bbc = ent->AddComponent<component::BoundingBoxComponent>(F_OBBFlags::COLLISION);
 		bbc->Init();
 		bbc->AddCollisionCategory<EnemyCollisionCategory>();
-		Physics::GetInstance().AddCollisionEntity(ent);
+		//Physics::GetInstance().AddCollisionEntity(ent);
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp]: Don't forget to remove this

File: StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp [hunk 2]
```diff
@@ -449,7 +454,8 @@ void EnemyFactory::onSceneSwitch(SceneChange* evnt)
 	if (evnt->m_NewSceneName == "ShopScene" || evnt->m_NewSceneName == "gameOverScene" || evnt->m_NewSceneName == "MainMenuScene")
 	{
 		m_IsActive = false;
-		m_Enemies.clear();
+		//m_Enemies.clear();
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/Gamefiles/EnemyFactory.cpp]: Don't forget to remove these

File: StortSpelProjekt/Game/src/main.cpp [hunk 3]
```diff
@@ -145,8 +145,7 @@ Scene* GameScene(SceneManager* sm)
 
 #pragma region assets
     AssetLoader* al = AssetLoader::Get();
-
-    al->GenerateMap(scene, "../Vendor/Resources/Rooms");
+    //al->GenerateMap(scene, "../Vendor/Resources/Rooms");
```
Comment (LeoWikstrom) [inline StortSpelProjekt/Game/src/main.cpp]: Don't forget to remove this

**Review by LeoWikstrom — CHANGES_REQUESTED** [review]
Fix spawn points for OutdoorsRoom

**Review by LeoWikstrom — DISMISSED** [review]

**Review by LeoWikstrom — DISMISSED** [review]

**Review by jocke1995 — CHANGES_REQUESTED** [review]
Remove SceneManager::AddEntity(), and use your new function (initDynamicEntity) instead.

File: StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp [hunk 4]
```diff
@@ -66,7 +66,7 @@ void component::EnemyComponent::OnUnInitScene()
     tc->GetTransform()->SetPosition(parentPos.x, parentPos.y, parentPos.z);
 
     entity->SetEntityState(true);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/EnemyComponent.cpp]: Now happens twice. This happens inside InitDynamicEntity. This function should be private now. And only be called from Scene::InitDynamicEntity

File: StortSpelProjekt/Game/src/Gamefiles/Components/RangeComponent.cpp [hunk 5]
```diff
@@ -236,7 +235,7 @@ void component::RangeComponent::Attack()
 
 		// add the entity to the sceneManager so it can be spawned in in run time
 		ent->SetEntityState(true);	// true == dynamic, which means it will be removed when a new scene is set
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/RangeComponent.cpp]: Same here

File: StortSpelProjekt/Game/src/Gamefiles/Components/RangeEnemyComponent.cpp [hunk 6]
```diff
@@ -155,7 +155,7 @@ void component::RangeEnemyComponent::Attack(float3 direction)
 
 		// add the entity to the sceneManager so it can be spawned in in run time
 		ent->SetEntityState(true);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/RangeEnemyComponent.cpp]: same here

File: StortSpelProjekt/Game/src/Gamefiles/Shop.cpp [hunk 7]
```diff
@@ -78,7 +78,7 @@ void Shop::Create2DGUI()
 		background
 	);
 	entity->SetEntityState(true);	// true == dynamic, which means it will be removed when a new scene is set
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Shop.cpp]: same here

File: StortSpelProjekt/Game/src/Gamefiles/UpgradeGUI.cpp [hunk 8]
```diff
@@ -197,7 +197,7 @@ void UpgradeGUI::CreateMenu(Scene* scene)
 			4,
 			notBlended,
 			m_pBoardBackgroundTexture, { 0.4, 0.4, 0.4 });
-		m_pSm->AddEntity(entity, scene);
+		scene->InitDynamicEntity(entity);
 		entity->Update(0);
 		entity->SetEntityState(true);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/UpgradeGUI.cpp]: same here

**Review by jocke1995 — CHANGES_REQUESTED** [review]
Fix pls

File: StortSpelProjekt/Game/src/Gamefiles/UpgradeGUI.cpp [hunk 9]
```diff
@@ -361,7 +361,7 @@ void UpgradeGUI::makeUpgradeButton(float2 pos, std::string name)
 		notBlended,
 		m_pButtonParchment);
 
-	m_pSm->AddEntity(entity, m_pCurrentScene);
+	m_pCurrentScene->InitDynamicEntity(entity);
 	entity->Update(0);
 	entity->SetEntityState(true);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/UpgradeGUI.cpp]: here :)

**Review by jocke1995 — APPROVED** [review]
good job! No problems at all, Excellent coding skills! :)

**Review by LeoWikstrom — APPROVED** [review]
👍🏽

**Pull request merged**